src/core/thread/meshcop_dataset.cpp

```diff
@@ -121,6 +121,20 @@ void Dataset::Get(otOperationalDataset &aDataset)
             break;
         }
 
+        case Tlv::kChannelMask:
+        {
+            const uint8_t *entry = reinterpret_cast<const uint8_t *>(cur) + sizeof(Tlv) + sizeof(ChannelMaskEntry);
+
+            aDataset.mChannelMaskPage0 = 0;
+            aDataset.mChannelMaskPage0 |= (entry[0] << 24);
+            aDataset.mChannelMaskPage0 |= (entry[1] << 16);
+            aDataset.mChannelMaskPage0 |= (entry[2] << 8);
+            aDataset.mChannelMaskPage0 |= entry[3];
```

Comment:
The channel mask is encoded in little-endian (least-significant bit of the first byte is channel 0).